sal/tests/dataclass/unit/test_common.py

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+import pytest
+
+from sal import dataclass
+
+DATACLASSES = (dataclass.DataObject, dataclass.DataSummary)
+
+
+@pytest.mark.parametrize('fixture, class_, group, version',
+    [('scalar', 'scalar', 'core', 1),
+     ('scalar_summary', 'scalar', 'core', 1),
+     ('string', 'string', 'core', 1),
+     ('string_summary', 'string', 'core', 1),
+     ('array_status', 'mask_array_status', 'signal_mask', 1),
+     ('array_status_summary', 'mask_array_status', 'signal_mask', 1),
+     ('scalar_status', 'mask_scalar_status', 'signal_mask', 1),
+     ('scalar_status_summary', 'mask_scalar_status', 'signal_mask', 1),
+     ('asym_array_error', 'error_asymmetrical', 'signal_error', 1),
+     ('asym_array_error_summary', 'error_asymmetrical', 'signal_error', 1),
+     ('constant_error', 'error_constant', 'signal_error', 1),
+     ('constant_error_summary', 'error_constant', 'signal_error', 1),
+     ('sym_array_error', 'error_symmetrical', 'signal_error', 1),
+     ('sym_array_error_summary', 'error_symmetrical', 'signal_error', 1),
+     ('array_dim', 'coordinate_array', 'signal_dimension', 1),
+     ('array_dim_summary', 'coordinate_array', 'signal_dimension', 1),
+     ('calc_dim', 'coordinate_calc', 'signal_dimension', 1),
+     ('calc_dim_summary', 'coordinate_calc', 'signal_dimension', 1),
+     ('signal', 'signal', 'signal', 1),
+     ('signal', 'signal', 'signal', 1)])
+def test_class_attributes(fixture, class_, group, version, request):
```

Comment:
**question:** `test_array.py` tests setting the description, was it decided that this was not necessary for the new tests?

Reply:
Yeah, it is either a simple assignment or a redundant conditional, which has been removed in another branch I'm working on.  Whether or not `description` is set correctly is also obliquely tested in the `to_dict` and `from_dict` tests, so it is covered anyway, albeit not in the way it would be if I thought it was necessary to test.